File: L2/src/database/connectMongo.ts
import { MongoClient, ServerApiVersion } from 'mongodb';

const uri = "mongodb+srv://joaoggbs62:gZ8PatNu0bI3lHOb@cluster0.wqjd7lm.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0";

// Criar um novo cliente MongoDB
const client = new MongoClient(uri, {
  serverApi: {
    version: ServerApiVersion.v1,
    strict: true,
    deprecationErrors: true,
  }
});

async function connectMongo() {
  try {
    await client.connect();
    return client;
  } catch (e) {
    console.error("Erro ao conectar ao MongoDB", e);
    throw e;
  }
}

export default connectMongo;

File: L3/database/connectMongo.py
from pymongo import MongoClient
from pymongo.errors import PyMongoError  # Importar a classe mais genérica

# URI de conexão com o MongoDB
uri = "mongodb+srv://joaoggbs62:gZ8PatNu0bI3lHOb@cluster0.wqjd7lm.mongodb.net/?retryWrites=true&w=majority"

def connect_mongo():
    """Conecta ao MongoDB e retorna o cliente."""
    client = MongoClient(uri)
    try:
        client.admin.command('ping')  # Testa a conexão
        print("Conectado ao MongoDB com sucesso!")
        return client
    except PyMongoError as e:  # Captura qualquer erro do PyMongo
        print("Erro ao conectar ao MongoDB:", e)
        raise e
